data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -73,23 +73,65 @@ _ "This attack uses 6 attacks"#enddef
     [/event]
 #enddef
 
-#define WEAPON_SPECIAL_SONG_VERSE ACTIVE_ON X DISTINGUISHER
+#define STR_VERSE_RESTRICT_1
+_ "This attack is the first verse of the Song of Sun Ascension, which progresses by one verse each turn this unit uses a song verse attack. When the last verse is sung or the song sequence is broken due to not attacking for a turn, the song starts again from the first verse."#enddef
+
+#define STR_VERSE_RESTRICT_2
+_ "This attack the second verse of the Song of Sun Ascension, which progresses by one verse each turn this unit uses a song verse attack. When the last verse is sung or the song sequence is broken due to not attacking for a turn, the song starts again from the first verse."#enddef
+
+#define STR_VERSE_RESTRICT_3
+_ "This attack the third verse of the Song of Sun Ascension, which progresses by one verse each turn this unit uses a song verse attack. When the last verse is sung or the song sequence is broken due to not attacking for a turn, the song starts again from the first verse."#enddef
+
+#define STR_VERSE_NAME_1
+_ "first verse"#enddef
+
+#define STR_VERSE_NAME_2
+_ "second verse"#enddef
+
+#define STR_VERSE_NAME_3
+_ "third verse"#enddef
+
+# param MAX is the total number of verses the unit has, needed for id
+
+#define WEAPON_SPECIAL_SONG_VERSE_BASE MAX
+    [disable]
+        id = verse_{MAX}
+        [filter_self]
+            formula = "self.wml_vars.current_verse > 1"
+        [/filter_self]
+        active_on = offense
+        name = {STR_VERSE_NAME_1}
+        description = {STR_VERSE_RESTRICT_1}
+    [/disable]
+#enddef
+
+# param MAX is the this attack should be active on
+# param MAX is the total number of verses the unit has, needed for id
+# example: {WEAPON_SPECIAL_SONG_VERSE 2 3} means this attack should be active on second verse out of three
+# don't set MAX to 1, use WEAPON_SPECIAL_SONG_VERSE_BASE instead to prevent an obscure error related to debug/script unit placement
```

Comment:
Having a blank line between this and the `#define` detaches the docs from the macro (for reading it manually, I'm not talking about automated parsing).

The coding style used in data/core/macros is to put the documentation inside the `#define`.